extensions/panache/hibernate-reactive-panache-common/runtime/src/main/java/io/quarkus/hibernate/reactive/panache/common/runtime/SessionOperations.java

```diff
@@ -26,31 +33,44 @@ public final class SessionOperations {
 
     private static final String ERROR_MSG = "Hibernate Reactive Panache requires a safe (isolated) Vert.x sub-context, but the current context hasn't been flagged as such.";
 
-    private static final LazyValue<Mutiny.SessionFactory> SESSION_FACTORY = new LazyValue<>(
-            new Supplier<Mutiny.SessionFactory>() {
-                @Override
-                public SessionFactory get() {
-                    // Note that Mutiny.SessionFactory is @ApplicationScoped bean - it's safe to use the cached client proxy
-                    Mutiny.SessionFactory sessionFactory = Arc.container().instance(Mutiny.SessionFactory.class).get();
-                    if (sessionFactory == null) {
-                        throw new IllegalStateException("Mutiny.SessionFactory bean not found");
-                    }
-                    return sessionFactory;
-                }
-            });
+    private static final ComputingCache<String, Key<Mutiny.Session>> SESSION_KEY_MAP = new ComputingCache<>();
+    private static final ComputingCache<String, Mutiny.SessionFactory> SESSION_FACTORY_MAP = new ComputingCache<>();
 
-    private static final LazyValue<Key<Mutiny.Session>> SESSION_KEY = new LazyValue<>(
-            new Supplier<Key<Mutiny.Session>>() {
+    private static SessionFactory createSessionFactory(String persistenceunitname) {
```

Comment:
This could be a function passed to `new ComputingCache(...)`? That way defaults are retrieved implicitly.